test/enforce-compiletime-test.cc

```diff
@@ -0,0 +1,137 @@
+// Formatting library for C++ - formatting library tests
+//
+// Copyright (c) 2012 - present, Victor Zverovich
+// All rights reserved.
+//
+// For the license information refer to format.h.
+
+#include <array>
+#include <iterator>
+#include <list>
+#include <map>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef WIN32
+#  define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include "fmt/chrono.h"
+#include "fmt/color.h"
+#include "fmt/format.h"
+#include "fmt/locale.h"
+#include "fmt/ostream.h"
+#include "fmt/ranges.h"
+
+// Exercise the API to verify that everything we expect to can compile.
+void TestFormatApi() {
+  (void)fmt::format(FMT_STRING("{}"), 42);
+  (void)fmt::format(FMT_STRING(L"{}"), 42);
+#if !FMT_GCC_VERSION
```

Comment:
Why is it conditionally compiled?

Reply:
Because it crashes in GCC if it's not - This is the issue I reported here: https://github.com/fmtlib/fmt/issues/2039. It seems to still be a problem: https://github.com/fmtlib/fmt/pull/2038/checks?check_run_id=1544063743